src/FirmwarePlugin/APM/APMFirmwarePlugin.cc

```diff
@@ -825,3 +827,84 @@ QString APMFirmwarePlugin::internalParameterMetaDataFile(Vehicle* vehicle)
         return QString();
     }
 }
+
+QString APMFirmwarePlugin::_getGitVersionUrl(Vehicle* vehicle)
+{
+    const static QString  baseUrl("http://firmware.ardupilot.org/%1/stable/PX4/git-version.txt");
+    if (vehicle->fixedWing()) {
+        return baseUrl.arg("Plane");
+    } else if (vehicle->vtol()) {
+        return baseUrl.arg("Plane");
+    } else if (vehicle->rover()) {
+        return baseUrl.arg("Rover");
+    } else if (vehicle->sub()) {
+        return baseUrl.arg("Sub");
+    }
+    return baseUrl.arg("Copter");
+}
+
+void APMFirmwarePlugin::checkIfIsLatestStable(Vehicle* vehicle)
+{
+    // This is required as mocklink uses a hardcoded firmware version
+    if (qgcApp()->runningUnitTests()) {
+        qCDebug(APMFirmwarePluginLog) << "Skipping version check";
+        return;
+    }
+    QString versionFile = _getGitVersionUrl(vehicle);
+    qCDebug(APMFirmwarePluginLog) << "Downloading" << versionFile;
+    QGCFileDownload* downloader = new QGCFileDownload(this);
+    connect(
+        downloader,
+        &QGCFileDownload::downloadFinished,
+        this,
+        [vehicle, this](QString remoteFile, QString localFile) {
+            _versionFileDownloadFinished(&remoteFile, &localFile, vehicle);
```

Comment:
What is that ?
The reference should be used in the lambda arguments.
`(QString& remoteFile, QString& localFile)`